### ISSUE TITLE
ERROR:    Exception in ASGI application

### ISSUE THREAD

**Issue by G-force78** [CASIA-IVA-Lab/FastSAM#137]
Using the gradio UI in collab after uploading a 1024x1024 image I get this message when prompting with text

```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 289, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.10/dist-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/usr/local/lib/python3.10/dist-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/usr/local/lib/python3.10/dist-packages/fastapi/routing.py", line 273, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.10/dist-packages/fastapi/routing.py", line 192, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.10/dist-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/usr/local/lib/python3.10/dist-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.10/dist-packages/gradio/routes.py", line 289, in api_info
    return gradio.blocks.get_api_info(config, serialize)  # type: ignore
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 518, in get_api_info
    serializer = serializing.COMPONENT_MAPPING[type]()
KeyError: 'dataset'
```

**Comment by Qperable** [comment]
me too, I used `pip install gradio_client==0.2.7` to solve this problem, but got some gradio problem else
`future: <Task finished name='tomcs5rs3nb_1' coro=<Queue.process_events() done, defined at C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py:342> exception=1 validation error for PredictBody
event_id
  Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing>
Traceback (most recent call last):
  File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 346, in process_events
    client_awake = await self.gather_event_data(event)
  File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 219, in gather_event_data
    data, client_awake = await self.get_message(event, timeout=receive_timeout)
  File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 448, in get_message
    return PredictBody(**data), True
  File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\pydantic\main.py", line 159, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
pydantic_core._pydantic_core.ValidationError: 1 validation error for PredictBody
event_id
  Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing`

**Comment by zdj-1995** [comment]
一样 无法使用

**Comment by Qiang-Feng98** [comment]
> me too, I used `pip install gradio_client==0.2.7` to solve this problem, but got some gradio problem else `future: <Task finished name='tomcs5rs3nb_1' coro=<Queue.process_events() done, defined at C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py:342> exception=1 validation error for PredictBody event_id Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict] For further information visit https://errors.pydantic.dev/2.1/v/missing> Traceback (most recent call last): File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 346, in process_events client_awake = await self.gather_event_data(event) File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 219, in gather_event_data data, client_awake = await self.get_message(event, timeout=receive_timeout) File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 448, in get_message return PredictBody(**data), True File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\pydantic\main.py", line 159, in __init__ __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__) pydantic_core._pydantic_core.ValidationError: 1 validation error for PredictBody event_id Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict] For further information visit https://errors.pydantic.dev/2.1/v/missing`

Did you solve it?

**Comment by G-force78** [comment]
No I'm afraid not

**Comment by fuguiYang1** [comment]
pip install gradio==3.39.0 maybe can solve it

**Comment by Qperable** [comment]
> > me too, I used `pip install gradio_client==0.2.7` to solve this problem, but got some gradio problem else `future: <Task finished name='tomcs5rs3nb_1' coro=<Queue.process_events() done, defined at C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py:342> exception=1 validation error for PredictBody event_id Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict] For further information visit https://errors.pydantic.dev/2.1/v/missing> Traceback (most recent call last): File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 346, in process_events client_awake = await self.gather_event_data(event) File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 219, in gather_event_data data, client_awake = await self.get_message(event, timeout=receive_timeout) File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\gradio\queueing.py", line 448, in get_message return PredictBody(**data), True File "C:\Users\46917\.conda\envs\FastSAM\lib\site-packages\pydantic\main.py", line 159, in __init__ __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__) pydantic_core._pydantic_core.ValidationError: 1 validation error for PredictBody event_id Field required [type=missing, input_value={'data': ['data:image/jpe...on_hash': 'tomcs5rs3nb'}, input_type=dict] For further information visit https://errors.pydantic.dev/2.1/v/missing`
> 
> Did you solve it?

I solved this problem using gradio 3.40.1 and successfully ran the demo program

**Comment by Qiang-Feng98** [comment]
> pip install gradio==3.39.0 maybe can solve it

yes, it can solve the problem! Thanks!
